buildchain/buildchain/vagrant.py

```diff
@@ -40,16 +42,53 @@ def mkdir_dot_vagrant() -> None:
         'uptodate': [True],
     }
 
+def task_vagrantupnoprov() -> types.TaskDict:
+    """Run `vagrant up` without provisioning a development environment."""
+    vagrantupnoprov = [config.VAGRANT, 'up']
+    vagrantupnoprov.extend(map(shlex.quote,
+        config.VAGRANT_UP_ARGS + tuple(('--no-provision',))))
+    return {
+        'actions': [doit.tools.LongRunning(' '.join(vagrantupnoprov))],
+        'file_dep': [constants.VAGRANT_SSH_KEY_PAIR],
+        'task_dep': [],
+        'uptodate': [False],
+    }
+
+def task_vagrantsnapshot() -> types.TaskDict:
+    """Snapshot development environment."""
+    vagrantsnap = [config.VAGRANT, 'snapshot', 'save', '--force', 'bootstrap', config.VAGRANT_SNAPSHOT_NAME]
+
+    return {
+        'actions': [doit.tools.LongRunning(' '.join(vagrantsnap))],
+        'file_dep': [],
+        'task_dep': ['vagrantupnoprov'],
+        'uptodate': [run_once]
+    }
+
+def task_vagrantreset() -> types.TaskDict:
+    """Restore development environment snapshot."""
+    vagrantrestore = [config.VAGRANT, 'snapshot', 'restore', 'bootstrap', config.VAGRANT_SNAPSHOT_NAME]
+
+    return {
+        'actions': [doit.tools.LongRunning(' '.join(vagrantrestore))],
```

Comment:
Is `LongRunning` needed here? I think it will work without it (if yes, remove it).

Same comment for the other places where you used `LongRunning`.

Reply:
It works, but we lose vagrant output:
With `LongRunning`:
```
$ ./doit.sh vagrantreset
.  vagrantreset
==> bootstrap: Forcing shutdown of VM...
==> bootstrap: Restoring the snapshot 'bootstrap.autosnapshot'...
==> bootstrap: Checking if box 'centos/7' version '1811.02' is up to date...
==> bootstrap: Resuming suspended VM...
==> bootstrap: Booting VM...
==> bootstrap: Waiting for machine to boot. This may take a few minutes...
    bootstrap: SSH address: 127.0.0.1:2222
    bootstrap: SSH username: vagrant
    bootstrap: SSH auth method: private key
==> bootstrap: Machine booted and ready!
==> bootstrap: Machine already provisioned. Run `vagrant provision` or use the `--provision`
==> bootstrap: flag to force provisioning. Provisioners marked to run always will still run.
$
```
Without:
```
$ ./doit.sh vagrantreset
.  vagrantreset
$
```

Reply:
If you want to see `stdout`, simply add `'verbosity': 2` in your dict (see [here](http://pydoit.org/tasks.html#verbosity)).

`LongRunning` does that indeed, but other things as well. So if you only want `stdout` then setting verbosity is better than using `LongRunning` :slightly_smiling_face: